File: Array/GoLangDsa/SlidingWindow/Leetcode413.go
package main

func numberOfArithmeticSlices(nums []int) int {

	if len(nums) < 3 {

		return 0
	}

	Left := 0
	Right := 2
	Count := 0

	for Right < len(nums) {

		if nums[Right]-nums[Right-1] == nums[Right-1]-nums[Right-2] {

			Count += (Right - Left) - 1
		} else {

			Left = Right - 1
		}
		Right++
	}

	return Count

}

File: Array/Leetcode413.cpp
#include<vector>

int numberOfArithmeticSlices(std::vector<int>& nums)
{
    int n = nums.size();
    int Initialptr {0};
    int Endptr {0};
    int len {0};
    int count {0};

    while (Endptr < n)
    {
        len = Endptr - Initialptr + 1;
        if (len >=3)
        {
            int diff1 = nums[Initialptr + 1] - nums[Initialptr];
            int diff2 = nums[Endptr] - nums[Endptr - 1];
            if (diff1 == diff2)
          {
            count++;
          }
        else
        {
            Initialptr = Endptr - 1;
        }
        }
        
        
        ++Endptr;
    }
    return count;
}